### PR TITLE
(fix): npm audit throws vulnerability and screenshot unit tests error

### PR DESCRIPTION
**Pull request by PeterNgTr**
## Motivation/Description of the PR
This PR aims to:
- Fix `npm audit` issue

```
                       === npm audit security report ===                        
                                                                                
# Run  npm install --save-dev documentation@12.1.4  to resolve 1 vulnerability
SEMVER WARNING: Recommended action is a potentially breaking change
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Denial of Service                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mem                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ documentation [dev]                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ documentation > yargs > os-locale > mem                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1084                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```
- Fix error when running `screenshot failed tests`

```
(node:18482) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'test' of undefined
    at /Users/thanhnguyen/Desktop/CodeceptJS/lib/plugin/screenshotOnFail.js:93:44
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
(node:18482) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)
(node:18482) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
## Checklist:
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by elukoyanov** [comment]
How doedmulti reports live eith this update?

**Comment by PeterNgTr** [comment]
> How does multi reports live eith this update?

That's possibly breaking things, so I would move that mocha version bump in another PR, thou!

File: lib/plugin/screenshotOnFail.js [hunk 1]
```diff
@@ -131,4 +131,16 @@ module.exports = function (config) {
       }
     }, true);
   });
+
+  function _getUUID(test) {
+    if (test.uuid) {
+      return test.uuid;
+    }
+
+    if (test.ctx) {
```
Comment (elukoyanov) [inline lib/plugin/screenshotOnFail.js]: do we need check for test: `test.ctx  && test.ctx.test`?

File: lib/plugin/screenshotOnFail.js [hunk 2]
```diff
@@ -131,4 +131,16 @@ module.exports = function (config) {
       }
     }, true);
   });
+
+  function _getUUID(test) {
```
Comment (elukoyanov) [inline lib/plugin/screenshotOnFail.js]: Do we need jsdoc and typings for this function?

You can make jsdoc and mark it as `@private`

File: lib/plugin/screenshotOnFail.js (hunk 1 above)
Comment (PeterNgTr) [reply lib/plugin/screenshotOnFail.js]: I'm not sure if we have the case `test.ctx` is not undefined and `test.ctx.test` is `undefined`?

File: lib/plugin/screenshotOnFail.js (hunk 2 above)
Comment (PeterNgTr) [reply lib/plugin/screenshotOnFail.js]: ah no need, it's just being used in this.

File: lib/plugin/screenshotOnFail.js [hunk 3]
```diff
@@ -131,4 +131,16 @@ module.exports = function (config) {
...
+    if (test.uuid) {
+      return test.uuid;
+    }
+
+    if (test.ctx) {
+      return test.ctx.test.uuid;
+    }
+
+    return Math.floor(new Date().getTime() / 1000);
```
Comment (Vorobeyko) [inline lib/plugin/screenshotOnFail.js]: Why you not used uuid4() ?

Comment (PeterNgTr) [reply lib/plugin/screenshotOnFail.js]: @Vorobeyko well, it's used before like that. I just moved it to separate method. Not sure if something is broken if we use `uuid4()`?

Comment (DavertMik) [reply lib/plugin/screenshotOnFail.js]: uuid4 is too long... better nanoid but that's an extra dependency

**Pull request merged**